src/fetchers/default-fetcher.nix

```diff
@@ -26,6 +27,8 @@
         then
           if l.isStorePath (l.concatStringsSep "/" (l.take 4 (l.splitString "/" source.path)))
           then source.path
+          else if l.pathExists (l.traceVal "${sourceRoot}/${source.path}")
```

Comment:
Why is the path checked if exists here? The sourceRoot should be used if `rootName` and `rootVersion` are `null` (so in the `else` branch, you can add a check for it there).

Reply:
Okay, then I guess I have misunderstood you, I'll change the conditions like you suggest.

IIUC in the case when `rootName` is set, the relative path should be working already (relativised to the `rootName` source) and if that then is relative, then only then it should be handled by what I added.

Reply:
> IIUC in the case when rootName is set, the relative path should be working already (relativised to the rootName source) and if that then is relative, then only then it should be handled by what I added. Right?

Basically yeah. If the `rootName` exists, then the `"${overriddenSources."${source.rootName}"."${source.rootVersion}"}/${source.path}"` code should be used. If not, the `sourceRoot` can be used for figuring out the absolute path.